compiler/x/codegen/OMRAheadOfTimeCompile.cpp

```diff
@@ -0,0 +1,142 @@
+   TR_FrontEnd *fe = comp->fe();
+   TR::CodeGenerator* cg = comp->cg();
+   // calculate the amount of memory needed to hold the relocation data
+
+   for (auto aotIterator = cg->getExternalRelocationList().begin(); aotIterator != cg->getExternalRelocationList().end(); ++aotIterator)
+      (*aotIterator)->addExternalRelocation(cg);
+
+   TR::IteratedExternalRelocation *r;
+   for (r = self()->getAOTRelocationTargets().getFirst();
+        r != NULL;
+        r = r->getNext())
+      {
+      self()->addToSizeOfAOTRelocations(r->getSizeOfRelocationData());
+      }
+
+   // now allocate the memory  size of all iterated relocations + the header (total length field)
+
+   // Note that when using the SymbolValidationManager, the well-known classes
+   // must be checked even if no explicit records were generated, since they
+   // might be responsible for the lack of records.
+
+   if (self()->getSizeOfAOTRelocations() != 0)
+      {
+      // It would be more straightforward to put the well-known classes offset
+      // in the AOT method header, but that would use space for AOT bodies that
+      // don't use the SVM. TODO: Move it once SVM takes over?
+      uintptr_t reloBufferSize =
+         self()->getSizeOfAOTRelocations() +OMR::AheadOfTimeCompile::SIZEPOINTER ;
+      uint8_t *relocationDataCursor = self()->setRelocationData(
+         fe->allocateRelocationData(self()->comp(), reloBufferSize));
+      // set up the size for the region
+      *(uintptr_t *)relocationDataCursor = reloBufferSize;
+      relocationDataCursor +=  OMR::AheadOfTimeCompile::SIZEPOINTER;
+      }
+    if (self()->getSizeOfAOTRelocations() != 0)
+      {
+      uint8_t *relocationDataCursor = self()->setRelocationData(fe->allocateRelocationData(self()->comp(), self()->getSizeOfAOTRelocations() + sizeof(uintptr_t )));
+      uint8_t * copy = relocationDataCursor;
+      // set up the size for the region
+      TR::RelocationRecordBinaryTemplate* groups = reinterpret_cast<
+      TR::RelocationRecordBinaryTemplate*> (relocationDataCursor);
+      TR::RelocationRuntime *reloRuntime =self()->reloRuntime();
+      TR::RelocationTarget *reloTarget = reloRuntime->reloTarget();
+      OMR::RelocationRecordGroup reloGroup(groups);
+
+      reloGroup.setSize(reloTarget,self()->getSizeOfAOTRelocations()+ OMR::AheadOfTimeCompile::SIZEPOINTER);
+            self()->addToSizeOfAOTRelocations(OMR::AheadOfTimeCompile::SIZEPOINTER);
+      relocationDataCursor +=OMR::AheadOfTimeCompile::SIZEPOINTER;
+      TR::IteratedExternalRelocation *s;
+      for (s = self()->getAOTRelocationTargets().getFirst();
+           s != NULL;
+           s = s->getNext())
+         {
+         s->setRelocationData(relocationDataCursor);
+         s->initializeRelocation(cg);
+         relocationDataCursor += s->getSizeOfRelocationData();
+         }
+      }
+}
+
+uint8_t* OMR::X86::AheadOfTimeCompile::initializeAOTRelocationHeader(TR::IteratedExternalRelocation *relocation)
```

Comment:
AOT feels redundant in this function name

Reply:
Hello @mstoodle, thank you for comment. This name was not designed by me. It is used across OpenJ9 codebase. Should anything be done about it?

Reply:
probably not here. can you add it to the list of deferred changes I asked you to collect and maintain, please?

Reply:
Done, marked as resolved

Reply:
Done, thank you!